pkg/services/ngalert/state/manager.go

```diff
@@ -284,29 +282,19 @@ func (st *Manager) Put(states []*State) {
 }
 
 // TODO: Is the `State` type necessary? Should it embed the instance?
-func (st *Manager) saveAlertStates(ctx context.Context, states ...StateTransition) (saved, failed int) {
```

Comment:
I'm not keen that this function never returns an error because the caller will never know if it succeeded, but we can merge this change and discuss that another time

Reply:
I wonder what we can do with the error other than logging it. I am ok with returning error If there is anything useful we can do with it.